platform/linux-generic/odp_schedule_scalable.c

```diff
@@ -1007,6 +1007,8 @@ static void schedule_order_lock(unsigned lock_index)
 		       monitor32(&rctx->rwin->olock[lock_index],
 				 __ATOMIC_ACQUIRE) != rctx->sn)
 			doze();
+	atomic_store_release(&rctx->rwin->lock_acquired, lock_index, false);
```

Comment:
Again, no need for fancy atomics here since you're already protected by the ordered lock itself. You're just remembering the `lock_index` you locked.